zne/extrapolation/extrapolator.py

```diff
@@ -122,3 +116,71 @@ def _validate_sigma(
         if sigma is None:
             sigma = [1] * default_size  # Note: zeros mean that the data is exact
         return self._validate_data(sigma)
+
+    def _cross_validate_data_sigma(
+        self,
+        x_data: tuple[float, ...],
+        y_data: tuple[float, ...],
+        sigma_x: tuple[float, ...],
+        sigma_y: tuple[float, ...],
+    ) -> None:
+        """Cross-validate data and sigmas."""
+        sizes = {len(x_data), len(y_data), len(sigma_x), len(sigma_y)}
+        if len(sizes) != 1:
+            raise ValueError(
+                "Invalid data, all inputs should be of the same size: "
+                f"{len(x_data) = }, {len(y_data) = }, {len(sigma_x) = }, {len(sigma_y) = }."
+            )
+
+    def _validate_min_points(
+        self,
+        x_data: tuple[float, ...],
+        y_data: tuple[float, ...],  # pylint: disable=unused-argument
```

Comment:
Same reason as before, the interface requires those inputs in the signature because a general routine may want to use them. In this simple case, however, they are not necessary so the arg stays unused (notice that we cannot modify the interface's signature without breaking the abstraction)